frontend/src/hooks/reducers/appStateReducer.ts

```diff
@@ -480,6 +489,71 @@ export function appStateReducer(state: AppState, action: AppStateAction): AppSta
       year.importedManures || [],
       year.derivedManures || [],
     );
+  } else if (action.type === 'UPDATE_SOIL_NITRATE_CREDIT') {
```

Comment:
Is there a reason why this isn't included in the field update action, since it's tied to fields? If you're worried about it being triggered too much, maybe you can add in a boolean to the action that tells the reducer whether or not to perform this update.

Reply:
You are correct. You had said that tables were going to be saved during this week so I rewrote a bit of the logic to accommodate switching to your method.